ir/ir-inline.h

```diff
@@ -88,20 +88,20 @@ void IR::Vector<T>::visit_children(Visitor &v) {
             i++;
             continue;
         }
-        if (auto l = dynamic_cast<const Vector *>(n)) {
+        if (auto l = n->template to<Vector<T>>()) {
```

Comment:
I suggest you replace `auto n = v.apply_visitor(*i)` above with `const IR::Node *n = v.apply_visitor(*i)`. That will resolve the ambiguity and there will be no reason for `template` here.